docs/Talkbuchet.js

```diff
@@ -0,0 +1,609 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * HOW TO SETUP:
+ * -----------------------------------------------------------------------------
+ * - Set the right values in "talkOcsApiUrl", "user" and "password" (a user must
+ *   be used; guests do not work).
+ * - If HPB clustering is enabled, set the token of a conversation in "token"
+ *   (otherwise leave empty).
+ * - Set whether to use audio, video or both in the "getUserMedia" call.
+ * - Set the desired numbers in "publishersCount" and
+ *   "subscribersPerPublisherCount" (in a regular call with N participants you
+ *   would have N publishers and N-1 subscribers).
+ *
+ * HOW TO RUN:
+ * -----------------------------------------------------------------------------
+ * - In the browser, log in the Nextcloud server (with the same user as in this
+ *   script).
+ * - Copy and paste the full script in the console of the browser to run it.
+ * - To run it again execute "closeConnections()" in the console; then you must
+ *   reload the page and copy and paste the script again.
+ *
+ * HOW TO ENABLE AND DISABLE THE MEDIA DURING A TEST:
+ * -----------------------------------------------------------------------------
+ * You can manually enable and disable the media during a test by copying and
+ * pasting in the browser console the following commands:
+ * - For audio:
+ * stream.getAudioTracks()[0].enabled = TRUE_OR_FALSE
+ * - For video:
+ * stream.getVideoTracks()[0].enabled = TRUE_OR_FALSE
+ *
+ * Note that you can only enable and disable the original media specified in the
+ * "getUserMedia" call.
+ *
+ * HOW TO CALIBRATE:
+ * -----------------------------------------------------------------------------
+ * The script starts as many publishers and subscripters for each publisher as
+ * specified, each one being a real and independent peer connection to Janus.
+ * Therefore, how many peer connections can be established with a single script
+ * run depends on the client machine, both its CPU and its network.
+ *
+ * It is first needed to "calibrate" the client by running it with different
+ * numbers of publishers and subscribers to check how many peer connections are
+ * supported before doing the actual stress test on Janus.
+ *
+ * It is most likely that the CPU or network of a single client will be
+ * saturated before Janus is. The script will write to the console when a peer
+ * could not be connected or if it was disconnected after being connected; if
+ * there is a large number of those messages or the CPU consumption is very high
+ * the client has probably reached its limit.
+ *
+ * Besides the messages written by the script itself you can manually check the
+ * connection state by copying and pasting in the browser console the following
```

Comment:
maybe expose those as global utility functions exposed by the main script that can be called directly ?

Reply:
Yes, let me add helper functions for some things (either here or in a follow up).